File: Word Break/Mine.cpp
class Solution {
public:
    bool wordBreak(string s, vector<string>& wordDict) {
        int len = s.size();
        vector<bool> dp(len, false);
        for (int i = 0; i < len; i++){
            for (int j = 0; j <= i; j++){
                if (j != i)
                    dp[i] = dp[j] && find (wordDict.begin(), wordDict.end(), s.substr(j+1, i-j)) != wordDict.end();
                else
                    dp[i] = find (wordDict.begin(), wordDict.end(), s.substr(0, i+1)) != wordDict.end();
                if (dp[i])
                    break;
            }
        }
        return dp[len-1];
    }
};

File: Problems/680. Valid Palindrome II/mine.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        # Find difference
        left, right = self.find_difference(s, 0, len(s) - 1)
        # Remove one character at left, find difference
        if left < right:
            leftRemoveLeft, rightRemoveLeft = self.find_difference(s, left + 1, right)
        else:
            return True
        # Remove right character at right, find difference
        if leftRemoveLeft < rightRemoveLeft:
            leftRemoveRight, rightRemoveRight = self.find_difference(s, left, right - 1)
        else:
            return True
        
        if leftRemoveRight >= rightRemoveRight:
            return True
        # Still have difference, return False
        return False
    
    def find_difference(self, s, left, right):
        while left < right:
            if s[left] == s[right]:
                left += 1
                right -= 1
            else:
                break
        
        return left, right
